lib/fsm/logger.go

```diff
@@ -42,31 +45,37 @@ type Logger struct {
 
 // Debug logs a debug message
 func (l *Logger) Debug(args ...interface{}) {
+	initQueue()
 	l.FieldLogger.Debug(args...)
-	err := l.Operator.CreateLogEntry(l.Key, l.makeLogEntry(
-		fmt.Sprint(args...), "debug"))
-	if err != nil {
-		l.FieldLogger.Error(trace.DebugReport(err))
+
+	select {
+	case logEntryC <- l.makeLogEntry(fmt.Sprint(args...), "debug"):
+	default:
+		l.FieldLogger.Debug("operation logger dropped message: ", fmt.Sprint(args...))
 	}
 }
 
 // Info logs an info message
 func (l *Logger) Info(args ...interface{}) {
+	initQueue()
 	l.FieldLogger.Info(args...)
-	err := l.Operator.CreateLogEntry(l.Key, l.makeLogEntry(
-		fmt.Sprint(args...), "info"))
-	if err != nil {
-		l.FieldLogger.Error(trace.DebugReport(err))
+
+	select {
+	case logEntryC <- l.makeLogEntry(fmt.Sprint(args...), "info"):
+	default:
+		l.FieldLogger.Info("operation logger dropped message: ", fmt.Sprint(args...))
 	}
 }
 
 // Warn logs a warning message
 func (l *Logger) Warn(args ...interface{}) {
+	initQueue()
 	l.FieldLogger.Warn(args...)
-	err := l.Operator.CreateLogEntry(l.Key, l.makeLogEntry(
-		fmt.Sprint(args...), "warn"))
-	if err != nil {
-		l.FieldLogger.Error(trace.DebugReport(err))
+
+	select {
+	case logEntryC <- l.makeLogEntry(fmt.Sprint(args...), "warn"):
+	default:
+		l.FieldLogger.Debug("operation logger dropped message: ", fmt.Sprint(args...))
```

Comment:
Nit: I think what Dima was getting at above is that we try to capitalize the first word of log messages, so "**O**peration logger...." here and in all others. Not sure we'll ever see this but better to be consistent :)